src/gululog_idx.erl

```diff
@@ -357,8 +387,123 @@ file_sync_close(Fd) ->
   ok = file:sync(Fd),
   ok = file:close(Fd).
 
+%% @private Delete index file.
+-spec truncate_delete_do([{logid(), {segid(), position()}}], dirname(), ?undef | dirname()) ->
+        [filename()].
+truncate_delete_do(DeleteList, Dir, BackupDir) ->
+  SegIdList = lists:usort([SegIdX || {_, {SegIdX, _}} <- DeleteList]),
```

Comment:
then this map is not necessary if you return {SegIdX > SegId, SegId} in filtermap.